### PR TITLE
Fix cargo outdated issue temporarily

### LINKED ISSUE #590: `cargo oudated` is failing on commit workflow
**Describe the bug**
"cargo outdated" is failing on commit CI workflow. Because of this the PRs can not be merged.

**To Reproduce**
Every commit pushed onto the thin-edge.io repo is causing this issue

**Expected behavior**
"cargo outdated" check should pass.

**Screenshots**
Find the error logs below

Run actions-rs/cargo@v1
  with:
    command: outdated
    use-cross: false
  env:
    CARGO_TERM_COLOR: always
/usr/share/rust/.cargo/bin/cargo outdated
thread 'main' panicked at 'package cache lock is not currently held, Cargo forgot to call `acquire_package_cache_lock` before we got to this stack frame', /usr/share/rust/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-0.57.0/src/cargo/util/config/mod.rs:1542:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Error: The process '/usr/share/rust/.cargo/bin/cargo' failed with exit code 101

**Environment (please complete the following information):**
 - OS [ubuntu]
 - Hardware [x86]
 - System-Architecture [github hosted runner]
 - thin-edge.io version [e.g. 0.4.2]

**Additional context**
The workaround is to remove the mandatory check on `cargo outdated` so that the PRs can be merged.

### PR DESCRIPTION
**Pull request by rina23q**
## Proposed changes

Fix #590.

Changes:
- Add steps to print the version of `cargo`; `cargo test` and `cargo build`.
- Add steps to print the version of plugins; all others, e.g. `cargo-msrv`.
- Add a step `cargo install cargo-outdated`. This brings the latest version to the workflow.

Future:
- The step `cargo install cargo-outdated` should be removed. Unfortunately, as of today, without install, the version `0.10.0` of `cargo-outdated` is used. That version panics.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/commit-workflow.yml [hunk 1]
```diff
@@ -105,6 +131,27 @@ jobs:
...
+
+      # In usual case, we don't need this step.
+      # There is an issue with cargo-outdated 0.10.0, therefore, explicitly need to install the latest version.
+      # This step takes a certain time, so it should be removed in the future.
+      - name: Cargo install outdated
+        uses: actions-rs/cargo@v1
+        # https://github.com/marketplace/actions/rust-cargo
+        with:
+          command: install
+          args: cargo-outdated
```
Comment (PradeepKiruvale) [inline .github/workflows/commit-workflow.yml:146]: Don't we need to specify which version needs to be installed?. Will this always pull and install the recent version?

**Review by PradeepKiruvale — APPROVED** [review]
LGTM, left one comment.

**Review by abelikt — APPROVED** [review]
lgtm

**Comment by rina23q** [comment]
I will try to use toolchain. It could revert this change later. However, I believe that this workflow must be fixed immediately, so I would merge it.

**Pull request merged**